extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +177,119 @@ private void restartJobs() {
+                    scheduledExecutor.schedule(dailyJob, midNightExpression);
+                    logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUID);
+
+                    // Startup Job
+                    LocalDateTime currentTime = LocalDateTime.now();
+                    Expression cronExpression = new CronExpression(createCronFromTemporal(currentTime));
+                    scheduledExecutor.schedule(dailyJob, cronExpression);
+                    logger.info("Scheduled astro job-startup-{} for thing {}", typeId, thingUID);
+
+                    // Repeat Scheduled Job every configured seconds
+                    currentTime = LocalDateTime.now();
+                    LocalDateTime futureTimeWithInterval = currentTime.plusSeconds(thingConfig.getInterval());
+                    Date start = Date.from(futureTimeWithInterval.atZone(ZoneId.systemDefault()).toInstant());
+                    if (isPositionalChannelLinked()) {
+                        Expression expression = new CronExpression(
+                                createCronForRepeatEverySeconds(thingConfig.getInterval()), start);
+                        Job positionalJob = new PositionalJob(thingUID);
+                        scheduledExecutor.schedule(positionalJob, expression);
+                        logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
+                                thingConfig.getInterval(), thingUID);
                     }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
                 }
+            } catch (ParseException ex) {
+                logger.error("{}", ex.getMessage(), ex);
+            } finally {
+                monitor.unlock();
             }
-        }, 2000, TimeUnit.MILLISECONDS);
+        }, DELAY, MILLISECONDS);
     }
 
     /**
      * Stops all jobs for this thing.
      */
     private void stopJobs() {
-        logger.debug("Stopping jobs for thing {}", getThing().getUID());
-        synchronized (schedulerLock) {
-            if (quartzScheduler != null) {
-                try {
-                    String thingUid = getThing().getUID().toString();
-                    for (JobKey jobKey : quartzScheduler.getJobKeys(jobGroupEquals(thingUid))) {
-                        logger.debug("Deleting astro {} for thing '{}'", jobKey.getName(), thingUid);
-                        quartzScheduler.deleteJob(jobKey);
-                    }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
-                }
+        ThingUID thingUID = getThing().getUID();
+        int jobsRemoved = 0;
+        logger.debug("Stopping Scheduled Jobs for thing {}", thingUID);
+        monitor.lock();
+        try {
+            if (scheduledExecutor != null) {
+                BlockingQueue<Runnable> queue = scheduledExecutor.getQueue();
```

Comment:
I have made necessary changes to adapt to the existing API of `ExpressionThreadPoolExecutor` to remove the jobs.